spec/facility_spec.rb

```diff
@@ -1,26 +1,212 @@
+    expect(@facility_1.register_vehicle(@cruz)).to eq(nil)
+    expect(@facility_2.register_vehicle(@cruz)).to eq(nil)
+  end
+
+  it "can create a registration date" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@cruz)
+    expect(@cruz.registration_date).to eq(Date.today)
+  end
+
+  it "collects $25 for antique vehicles" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@camaro)
+
+    expect(@facility_1.collected_fees).to eq(25)
+  end
+
+  it "collects $200 for electric vehicles" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@bolt)
+
+    expect(@facility_1.collected_fees).to eq(200)
+  end
+
+  it "collects $100 for regular vehicles" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@cruz)
+
+    expect(@facility_1.collected_fees).to eq(100)
+  end
+
+  it "will return sum of collected_fees after multiple registrations" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@cruz)
+    @facility_1.register_vehicle(@bolt)
+    @facility_1.register_vehicle(@camaro)
+
+    expect(@facility_1.collected_fees).to eq(325)
+  end
+
+  it "stores all registered vehicles" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@cruz)
+    @facility_1.register_vehicle(@bolt)
+    @facility_1.register_vehicle(@camaro)
+
+    expect(@facility_1.registered_vehicles).to eq([@cruz, @bolt, @camaro])
+  end
+
+  it "can sort vehicles into plate types once registered" do
+    @facility_1.add_service("Vehicle Registration")
+    @facility_1.register_vehicle(@cruz)
+    @facility_1.register_vehicle(@bolt)
+    @facility_1.register_vehicle(@camaro)
+
+    expect(@cruz.plate_type).to eq(:regular)
+    expect(@bolt.plate_type).to eq(:ev)
+    expect(@camaro.plate_type).to eq(:antique)
+  end
+
+  it "will assign nil plate_type for unregistered vehicles" do
+    @facility_1.add_service("Vehicle Registration")
+
+    expect(@cruz.plate_type).to eq(nil)
+  end
+
+  it "can administer written test" do
+    @facility_1.add_service("Written Test")
+
+    expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+  end
+
+  it "will only administer test if facility has service" do
+    expect(@facility_1.administer_written_test(@registrant_1)).to eq(false)
+  end
+
+  it "will update license data after written test" do
+    @facility_1.add_service("Written Test")
+    @facility_1.administer_written_test(@registrant_1)
+
+    expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+  end
+
+  it "won't allow registrant to take written test if they don't have a permit" do
+    @facility_1.add_service("Written Test")
+    @facility_1.administer_written_test(@registrant_2)
+
+    expect(@registrant_2.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+  end
```

Comment:
Yeah, good job testing all the conditions for these methods. Since there are multiple conditions, we want to ensure we're testing all combinations of these conditions to ensure our methods function as expected.